@xen-orchestra/fs/src/azure.js

```diff
@@ -0,0 +1,195 @@
+import { BlobServiceClient } from '@azure/storage-blob'
+import { createLogger } from '@xen-orchestra/log'
+import { parse } from 'xo-remote-parser'
+import { basename, join, split } from './path'
+import RemoteHandlerAbstract from './abstract'
+import { pRetry } from 'promise-toolbox'
+
+createLogger('xo:fs:azure')
+const MAX_BLOCK_SIZE = 1024 * 1024 * 1024 * 1024 * 4
+const MAX_BLOCK_COUNT = 1250
+const { warn } = createLogger('xo:fs:azure')
+
+// <http|https>://<account-name>.<service-name>.core.windows.net/<resource-path>
+
+export default class AzureHandler extends RemoteHandlerAbstract {
+  #container
+  #dir
+  #blobServiceClient
+  #containerClient
+
+  constructor(remote, _opts) {
+    super(remote)
+    const { username, path, password, host, protocol } = parse(remote.url)
+    /* 
+For Azure not Azurite
+ --------------------
+  const credentials = new StorageSharedKeyCredential(username, password);
+  this.#blobServiceClient = new BlobServiceClient(`${protocol}://${username}.blob.core.windows.net`, credentials);
+ --------------------
+*/
+
+    // this works ONLY for Azurite
+    this.#blobServiceClient = BlobServiceClient.fromConnectionString(
+      `DefaultEndpointsProtocol=${protocol};AccountName=${username};AccountKey=${password};BlobEndpoint=${protocol}://${host}/${username}`
```

Comment:
I think you should check if there is a host to connect to a azurite or an azure server. In this case please add a info() stating which connection method is used